### DISCUSSION TITLE
Better error messaging when a Streamfield fails block_counts restriction

### DISCUSSION THREAD

**Discussion by enzedonline** [wagtail/wagtail#11239] [Ideas]
Currently, if a page or snippet withstreamfield that fails a block_counts restriction, there's a generic error banner displayed at the top of the page but no indicator to say where the error is, no indicator in the mini-map.

For example, the following StreamBlock:

```python
class MenuStreamBlock(StreamBlock):
    page_link = PageLinkMenuBlock()
    url_link = URLLinkMenuBlock()
    autofill_menu = AutoFillMenuBlock()
    sub_menu = SubMenuBlock()
    search_form = SearchMenuBlock(max_num=1)
    user_menu = UserMenuBlock(max_num=1)

    class Meta:
        block_counts = {
            'search_form': {'min_num': 0, 'max_num': 1},
            'user_menu': {'min_num': 0, 'max_num': 1},
        }
```

A 2nd SearchMenuBlock has been added to the stream and the editor attempts to save a draft:

![image](https://github.com/wagtail/wagtail/assets/72310199/771f6937-aff5-45a1-a0a3-e17ef570395b)

The editor doesn't know what the error is or where to look for it and needs to scroll down the page to look for any error messages. 

At the top of the Streamfield, there is a banner indicating the error. It's formatted with the critical message (help-critical) style:

![image](https://github.com/wagtail/wagtail/assets/72310199/7dae3ac9-cbd2-4ba3-b08b-1b71d171fafc)

This doesn't really stand out to me, particularly on light mode and particularly if the streamfield error message is below the first fold. I tried a bit of ux testing with editors - they were all a bit confused and failed to find the error without some lengthy search.

1st improvement would be to mark the streamfield with the error on the minimap as per standard field errors:

![image](https://github.com/wagtail/wagtail/assets/72310199/dd7266fb-3da7-466f-9716-3526dab7d1fc)

Red is the colour used for other error messages, it's the colour the editor associates with errors when scanning the page.

![image](https://github.com/wagtail/wagtail/assets/72310199/139d2407-3a7b-4698-b2de-5979691ec8d3)

I think the error message should have the error banner styling to stand out and make the message more identifiable. Not sure if this means the critical message style needs a review, or if this is a case for an extra style.

![image](https://github.com/wagtail/wagtail/assets/72310199/82b42151-610e-4256-9eaf-db1c5dba7bb1)

**Comment by lb-** [comment]
Is this related to https://github.com/wagtail/wagtail/issues/10395 ?

**Reply by enzedonline** [reply]
Looks like it is the same issue for the minimap.
